### PR TITLE
[macOS] Title bar style

### PR DESCRIPTION
**Pull request by edipox**
Already tested in Linux, macOS and browser. It works perfectly because it only affects the macOS app.

keeweb_new
keeweb

### CODE REVIEW

File: app/templates/menu/menu.hbs [hunk 1]
```diff
@@ -1,2 +1,2 @@
-<div class="menu">
-</div>
\ No newline at end of file
+<div class="menu {{platform}}">
```
Comment (antelle) [inline app/templates/menu/menu.hbs]: Please pass it in app-view, here: https://github.com/keeweb/keeweb/blob/develop/app/scripts/views/app-view.js#L104
We don't need a helper, to use it only one time.

**Comment by antelle** [comment]
Cool, thanks for that! 👍 
Please make it as an option. It might disturb somebody, personally I like mac header more.

**Comment by edipox** [comment]
Ok, thanks @antelle I'll take a look at how to make it as an option.

**Comment by antelle** [comment]
It should be somewhere here: https://github.com/keeweb/keeweb/blob/master/app/scripts/views/settings/settings-general-view.js#L84
, and a checkbox on settings page.
In Atom it's called 'Use custom title-bar', you can put this name to locales/base.json as well.
Then read app-settings.json inside app.js and apply it.

**Comment by EnricoMonese** [comment]
I'm implementing the setting as a dropdown from which you can choose:
- default
- hidden
- hidden-inset
These are the options given from electron [here](https://github.com/electron/electron/blob/master/docs/api/browser-window.md)

I then intend to pass it here https://github.com/keeweb/keeweb/blob/develop/desktop/app.js#L114, what would be the best way to get the value from _app-settings.json_ from there?

**Comment by antelle** [comment]
@EnricoMonese just read app-settings.json there.
I guess, there's no possibility to change it in a running app, right? Then it's easier, you can just write in a label, that it will be applied only after restart.

**Comment by EnricoMonese** [comment]
Yes it can only be set when the window is created.

**Comment by antelle** [comment]
Cool, let's just mention it in the label then.

**Comment by EnricoMonese** [comment]
By reading it do you mean open the file using fs and then parsing it, or is there already something that can be imported that wraps this funcionality?

**Comment by antelle** [comment]
@EnricoMonese yes, just like window position:
https://github.com/keeweb/keeweb/blob/develop/desktop/app.js#L16
https://github.com/keeweb/keeweb/blob/develop/desktop/app.js#L208
But in your case, as we need this before we can create a window, you can just load it sync and parse.

**Comment by EnricoMonese** [comment]
Did something fast but it throws some errors. Will take a better look and commit later as now I have to go. Probably just forgot a comma somewhere though. I also saw that there is something like sheduleRestart could that be used after changing the setting?

**Comment by antelle** [comment]
Yes, it's used during updates, to trigger auto-update, you can use `Launcher.requestRestart();`.
If you want, you can add a button 'Restart me', which will appear after you change the mode and call this code from it.

**Comment by EnricoMonese** [comment]
Should I open a new pr?

**Comment by antelle** [comment]
@EnricoMonese, yes, I guess, you cannot modify this PR, the only way is to create a new one.

**Comment by EnricoMonese** [comment]
Also having the different titlebar means that there no longer is a draggable region. I made the whole window draggable from everywhere as explained [here](https://github.com/electron/electron/blob/master/docs/api/frameless-window.md#draggable-region), but I guess that maybe that should be an option too

**Comment by antelle** [comment]
@EnricoMonese hm, I can drag Atom in Custom Titlebar mode, I think it makes sense to make title region draggable, otherwise it's not usable at all.

**Comment by EnricoMonese** [comment]
Will add a checkbox to make the whole window draggable, if set to false it will use either the default titlebar or just the top of the content when the titlebar is hidden.

**Comment by antelle** [comment]
No, we don't need two settings just for this simple switch. Let's just make the title draggable? Are there any apps which are draggable by the whole window?

**Comment by EnricoMonese** [comment]
There are some but it seems a common behavior in apps with the hidden inset titlebar. As I already did the change I can say that it actually isn't too distracting. It just means that you can also drag the window by clicking on blank areas. I personally find it useful in other apps so I just thought that since it has to be addressed anyway we could simply add a toggle.

**Comment by antelle** [comment]
Let's add a draggable option to the bottom of the select.

**Comment by antelle** [comment]
Moved to #571

**Pull request closed without merging**